policyengine_api/routes/household_routes.py

```diff
@@ -0,0 +1,163 @@
+from flask import Blueprint, Response, request, jsonify
+from werkzeug.exceptions import NotFound, InternalServerError, BadRequest
+import json
+
+from policyengine_api.constants import COUNTRY_PACKAGE_VERSIONS
+from policyengine_api.services.household_service import HouseholdService
+from policyengine_api.utils import hash_object
+from policyengine_api.utils.payload_validators import (
+    validate_household_payload,
+    validate_country,
+)
+
+
+household_bp = Blueprint("household", __name__)
+household_service = HouseholdService()
+
+
+@household_bp.route(
+    "/<country_id>/household/<int:household_id>", methods=["GET"]
+)
+@validate_country
+def get_household(country_id: str, household_id: int) -> Response:
+    """
+    Get a household's input data with a given ID.
+
+    Args:
+        country_id (str): The country ID.
+        household_id (int): The household ID.
+    """
+    print(f"Got request for household {household_id} in country {country_id}")
+
+    try:
+        household: dict | None = household_service.get_household(
+            country_id, household_id
+        )
+        if household is None:
+            raise NotFound(f"Household #{household_id} not found.")
+        else:
+            return Response(
+                json.dumps(
+                    {
+                        "status": "ok",
+                        "message": None,
+                        "result": household,
+                    }
+                ),
+                status=200,
+                mimetype="application/json",
+            )
+    except NotFound:
+        raise
+    except Exception as e:
+        raise InternalServerError(
+            f"An error occurred while fetching household #{household_id}. Details: {str(e)}"
+        )
```

Comment:
**suggestion, nonblocking** I would suggest you don't need to do this at all. Just let the top-level exception handler deal with it.

The client knows what request they were executing so they should know the household was already and the server should log the request so your logs should have which request too. 

And the cost is an extra onion layer of exception handling in every route function.

If you _do_ want to do it though, I would catch all the HTTPException objects and reraise first instead of each by name (what if you forget one?)
```
except HTTPException:
  raise
except Exception as e:
...
```

Reply:
I've refactored against an implementation using `Blueprint` and have just instead imported the error itself and returned on-the-spot when necessary. I've also removed the specific exception handlers.